uaclient/contract.py

```diff
@@ -105,6 +106,22 @@ def request_resources(self) -> "Dict[str, Any]":
         )
         return resource_response
 
+    def request_premium_aws_attach(self, pkcs7: str):
+        """Requests machine attach for premium images on AWS.
+
+        @param pkcs7: string obtained from AWS metadata service from
+            http://169.254.169.254/latest/dynamic/instance-identity/pkcs7
+            from this instance.
+
+        @return: Dict of the JSON response containing the contract-token.
+        """
+        data = {"pkcs7": pkcs7}
+        response, _headers = self.request_url(
+            API_V1_PREMIUM_AWS_TOKEN, data=data
```

Comment:
We aren't setting `content-type` here, in what form does this end up getting sent over the wire?

Reply:
well, the same utf-encoded byte-string. if no headers are provided, self.headers() is called which the base serviceclient defines as
    def headers(self):
        return {
            "user-agent": "UA-Client/{}".format(version.get_version()),
            "accept": "application/json",
            "content-type": "application/json",
        }

As a result, the headers settings will say we are encoding json content and so we request_url encodes a json string. We probably need to clean up the unnecessary headers being set.